agent/azure-monitor-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExportersEndToEndTest.java

```diff
@@ -0,0 +1,244 @@
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.GZIPInputStream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import reactor.core.publisher.Mono;
+import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
+import uk.org.webcompere.systemstubs.jupiter.SystemStub;
+import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension;
+
+@ExtendWith(SystemStubsExtension.class)
+public class AzureMonitorExportersEndToEndTest extends MonitorExporterClientTestBase {
+
+  @SystemStub EnvironmentVariables envVars = new EnvironmentVariables();
+
+  private static final String TRACE_CONNECTION_STRING =
+      "InstrumentationKey=00000000-0000-0000-0000-000000000000";
+  private static final String INSTRUMENTATION_KEY = "00000000-0000-0000-0000-0FEEDDADBEEF";
+
+  @BeforeEach
+  public void setup() {
+    envVars.set(
+        "APPLICATIONINSIGHTS_CONNECTION_STRING", "InstrumentationKey=" + INSTRUMENTATION_KEY);
+  }
+
+  @Test
+  public void testBuildMetricExporter() throws Exception {
+    validateMetricExporterEndToEnd("testBuildMetricExporter");
+  }
+
+  @Test
+  public void testBuildTraceExporter() throws Exception {
+    validateTraceExporterEndToEnd("testBuildTraceExporter");
+  }
+
+  // OpenTelemetry doesn't have a Log API
+  @Test
+  public void testBuildLogExporter() throws Exception {
+    validateLogExporterEndToEnd();
+  }
+
+  @Test
+  public void testBuildTraceMetricLogExportersConsecutively() throws Exception {
+    validateTraceExporterEndToEnd("testBuildTraceMetricLogExportersConsecutively");
+    validateMetricExporterEndToEnd("testBuildTraceMetricLogExportersConsecutively");
+    validateLogExporterEndToEnd();
+  }
+
+  private static void validateMetricExporterEndToEnd(String testName) throws Exception {
+    CustomValidationPolicy customValidationPolicy = generateMetrics(testName);
+    List<TelemetryItem> actualTelemetryItems = customValidationPolicy.getActualTelemetryItems();
+    assertThat(actualTelemetryItems.size()).isGreaterThan(0);
+    TelemetryItem actualTelemetryItem = actualTelemetryItems.get(0);
+    assertThat(actualTelemetryItem.getName()).isEqualTo("Metric");
+    assertThat(actualTelemetryItem.getInstrumentationKey()).isEqualTo(INSTRUMENTATION_KEY);
+    assertThat(actualTelemetryItem.getTags())
+        .hasEntrySatisfying("ai.cloud.role", v -> assertThat(v).isEqualTo("unknown_service:java"));
```

Comment:
this can be simplified
```suggestion
    assertThat(actualTelemetryItem.getTags())
        .containsEntry("ai.cloud.role", "unknown_service:java");
```